### DISCUSSION TITLE
After Update per Docker can't reach the web-interface anymore. :(

### DISCUSSION THREAD

**Discussion by jojo-gfx** [immich-app/immich#11087] [Q&A] [answered]
Hi,

was yesterday updating my immich installation on Ubuntu 22.04.4 and all was well and all containers are running after it.
But I can't connect to my server now per web-interface, so I have a blackbox now. :(
What could be wrong here?
Any idea?

Greetings :)

**Comment by bo0tzz** [comment]
We can't help you without any information ;) Please post at least the logs, and make sure all the containers are running etc.

**Comment by jojo-gfx** [comment]
The system-log orr are there also other important logs?
All containers are running.

**Comment by jojo-gfx** [comment]
This is often list in last system log:


Jul 14 07:12:07 fotos containerd[125]: time="2024-07-14T07:12:07.212476440Z" level=info msg="shim disconnected" id=c792eb4a2be4f4ffc16c435ee17a6510fc7b4a95be8830cb297be0d2e333c13c namespace=moby
Jul 14 07:12:07 fotos containerd[125]: time="2024-07-14T07:12:07.212548394Z" level=warning msg="cleaning up after shim disconnected" id=c792eb4a2be4f4ffc16c435ee17a6510fc7b4a95be8830cb297be0d2e333c13c namespace=moby
Jul 14 07:12:07 fotos containerd[125]: time="2024-07-14T07:12:07.212561598Z" level=info msg="cleaning up dead shim" namespace=moby
Jul 14 07:12:07 fotos systemd-networkd[82]: veth875717d: Lost carrier
Jul 14 07:12:07 fotos networkd-dispatcher[103]: WARNING:Unknown index 47 seen, reloading interface list
Jul 14 07:12:07 fotos systemd-networkd[82]: veth875717d: Link DOWN
Jul 14 07:12:07 fotos networkd-dispatcher[103]: ERROR:Unknown interface index 47 seen even after reload
Jul 14 07:12:07 fotos networkd-dispatcher[103]: WARNING:Unknown index 47 seen, reloading interface list
Jul 14 07:12:07 fotos networkd-dispatcher[103]: ERROR:Unknown interface index 47 seen even after reload
Jul 14 07:12:07 fotos networkd-dispatcher[103]: WARNING:Unknown index 47 seen, reloading interface list
Jul 14 07:12:07 fotos networkd-dispatcher[103]: ERROR:Unknown interface index 47 seen even after reload
Jul 14 07:12:07 fotos networkctl[4920]: Interface "veth875717d" not found.
Jul 14 07:12:07 fotos systemd[1]: networkd-dispatcher.service: Got notification message from PID 4920, but reception only permitted for main PID 103
Jul 14 07:12:07 fotos networkd-dispatcher[103]: ERROR:Failed to get interface "veth875717d" status: Command '['/usr/bin/networkctl', 'status', '--no-pager', '--no-legend', '--', 'veth875717d']' returned non-zero exit status 1.
Jul 14 07:12:07 fotos systemd[1]: Reloading Postfix Mail Transport Agent (instance -)...
Jul 14 07:12:07 fotos systemd[1]: run-docker-netns-3470369b564e.mount: Deactivated successfully.
Jul 14 07:12:07 fotos systemd[1]: var-lib-docker-overlay2-4acbaf7ba652204565c30d82ba4745c8eff5359223750f7b9e14dba0e5326fae-merged.mount: Deactivated successfully.
Jul 14 07:12:07 fotos postfix[4934]: Postfix is running with backwards-compatible default settings
Jul 14 07:12:07 fotos postfix[4934]: See http://www.postfix.org/COMPATIBILITY_README.html for details
Jul 14 07:12:07 fotos postfix[4934]: To disable backwards compatibility use "postconf compatibility_level=3.6" and "postfix reload"
Jul 14 07:12:07 fotos postfix/postfix-script[4940]: refreshing the Postfix mail system
Jul 14 07:12:07 fotos postfix/master[416]: reload -- version 3.6.4, configuration /etc/postfix
Jul 14 07:12:07 fotos systemd[1]: Reloaded Postfix Mail Transport Agent (instance -).
Jul 14 07:12:07 fotos systemd[1]: Reloading Postfix Mail Transport Agent...
Jul 14 07:12:07 fotos systemd[1]: Reloaded Postfix Mail Transport Agent.
Jul 14 07:12:07 fotos networkd-dispatcher[103]: WARNING:Unknown index 47 seen, reloading interface list
Jul 14 07:12:07 fotos networkd-dispatcher[103]: ERROR:Unknown interface index 47 seen even after reload
Jul 14 07:12:09 fotos systemd[1]: docker-3815425e6d97b4ba72cf9d90d4ae7a0b996a95de99d2abe132b0a37167e61a78.scope: Deactivated successfully.
Jul 14 07:12:09 fotos systemd[1]: docker-3815425e6d97b4ba72cf9d90d4ae7a0b996a95de99d2abe132b0a37167e61a78.scope: Consumed 6.665s CPU time.
Jul 14 07:12:09 fotos dockerd[149]: time="2024-07-14T07:12:09.970157488Z" level=info msg="ignoring event" container=3815425e6d97b4ba72cf9d90d4ae7a0b996a95de99d2abe132b0a37167e61a78 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskD
elete"

**Comment by jojo-gfx** [comment]
Oh, on console is shown that:
root@fotos:~/immich-app# docker compose up -d
WARN[0000] /root/immich-app/docker-compose.yml: `version` is obsolete
[+] Running 5/0
 ✔ Container immich_machine_learning  Running                                                                                                                                                                                                              0.0s
 ✔ Container immich_redis             Running                                                                                                                                                                                                              0.0s
 ✔ Container immich_postgres          Running                                                                                                                                                                                                              0.0s
 ✔ Container immich_microservices     Started                                                                                                                                                                                                              0.0s
 ✔ Container immich_server            Started   #11070 


..... have I to use a new yml? Why it isn't replaced on update?

**Comment by jojo-gfx** [comment]
With new .yml file the variable "DB_DATA_LOCATION" is missed.
Seems the new version has here a new variable who is on update ignored ... same like with the .yml file ...  :(

What should I put in the .env file for that?

... ok, have insert in the .env the following line:
DB_DATA_LOCATION=/var/lib/postgresql/data

System is running now, but what a horor! 
Seems all Data is away ... I'm the first user on the system and should register me. ;(
How can that be? The database is the same!
Is here the whole table structure wrong now for the new version?
Can't take a look into the Database, because it is a blackbox in a container for me and cant connect from other computers.

With making a new account is no data available in immich.
Menu means 500GB of 800 are filled, but could be only an ordinary system output for free space on SSD.

Seems the data itself is still there in the data folder, but without the metadata in the database it seems to be stupid senseless data.

Can I do here anything or is all lost now?

**Comment by jojo-gfx** [comment]
No idea? :(

**Reply by alextran1502** [reply]
Can you put back the pg_data volume?

**Reply by jojo-gfx** [reply]
Have no idea, what I have to do there. The whole concept of docker and data in this system is a black hole for me... 
Could be the path in the version before was another one? But where was it before?
It was only a workaround with what I've found in docs.

**Comment by jojo-gfx** [comment]
Have deleted now the server and use a post version without my data ... try "docker compose pull && docker compose up -d" ... and also now the immich isn't working anymore or can reach the web-frontend ... dont know what is wrong. :( 
wtf!?

**Reply by jojo-gfx** [reply]
Ah, again: WARN[0000] /root/immich-app/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion

The update doesn't correct this file and so all goes wrong. :(
I try to fix it, but think it will kill the database again.

The developers forget here to update the yml-file! Beware to uptate a version from 2023!
It's never a good idea to change the structure of such files without essential need! :(

**Reply by jojo-gfx** [reply]
.......... I give up!

**Reply by jojo-gfx** [reply, marked as answer]
Ok, also complete new installation goes wrong same way, so I changed the platform:
-> Don't use Ubuntu 22! Seems it isn't compatible at time with the newest version.
-> I have install it now on Debian12 and instantly all is running perfect. o.O
